source/common/upstream/ring_hash_lb.cc

```diff
@@ -136,12 +138,36 @@ RingHashLoadBalancer::Ring::Ring(const Optional<envoy::api::v2::Cluster::RingHas
       config.valid()
           ? PROTOBUF_GET_WRAPPED_OR_DEFAULT(config.value().deprecated_v1(), use_std_hash, true)
           : true;
+
+  std::unique_ptr<char[]> hash_key_buffer;
+  uint64_t last_hash_key_size = 0;
   for (const auto& host : hosts) {
+    const std::string& address_string = host->address()->asString();
+    uint64_t offset_start = address_string.size();
+
+    // Although in almost all cases the buffer could be stack allocated to 128 bytes or so, we
+    // don't explicitly know what is in the address string (e.g., UDS path). We allocate a raw
+    // buffer on the heap and use it for all calculations. The needed size is the size of the
+    // address, plus '_', plus 32 bytes for the index. All of this is done to avoid string
+    // allocations in the fast path.
+    const uint64_t needed_size = std::min(128UL, address_string.size() + 1 + 32);
```

Comment:
Is there some constant we can use for max 64-bit representation? In the `StringUtil::itoa` comment it says this is 21 (including null), here we use 32 just to be safe (I think).

Reply:
Yeah I will add a constant. I realized this code has a bug, it should be `std::max`, not min. I'm going to redo it also to use https://github.com/abseil/abseil-cpp/blob/master/absl/container/inlined_vector.h which would make it stack allocated in most case and easier to read. I will update all of it tomorrow.